.github/workflows/ccpp.yml

```diff
@@ -1,19 +1,22 @@
-name: CCPP CI
+name: C/C++ CI
 
 on: [push]
 
 jobs:
   build:
-
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-18.04
 
     steps:
-    - uses: actions/checkout@v1
-      with:
-        submodules: recursive
-    - name: prepare
+    - uses: actions/checkout@v2
+    - name: Checkout submodules
+      shell: bash
+      run: |
+        auth_header="$(git config --local --get http.https://github.com/.extraheader)"
+        git submodule sync --recursive
+        git -c "http.extraheader=$auth_header" -c protocol.version=2 submodule update --init --force --recursive --depth=1
```

Comment:
what's the difference to the one on the left? @nevrending ?

Reply:
actions@v2 removed the `submodules` input
https://github.com/actions/checkout#Checkout-submodules